common/src/main/kotlin/org/wycliffeassociates/otter/common/domain/content/CreateChunks.kt

```diff
@@ -76,22 +85,140 @@ class CreateChunks(
         writeChunkFile(projectSlug, chapterNumber, chunksToAdd)
     }
 
+    /**
+     * Creates chunks from the verses in the text of the project.
+     *
+     * @param versificationRepository the versification repository to load the appropriate versification to preallocate
+     * @param projectSlug the slug of the project to create chunks for
+     * @param draftNumber the draft number to create chunks for
+     */
     fun createChunksFromVerses(
+        versificationRepository: IVersificationRepository,
         projectSlug: String,
         draftNumber: Int
     ) {
-        val chunksToAdd = mutableListOf<Content>()
+        ResourceContainer
+            .load(workbook.source.resourceMetadata.path).use { rc ->
+                val vrsSlug = rc.manifest.projects.firstOrNull { !it.versification.isNullOrEmpty() }?.versification ?: ""
+                versificationRepository.getVersification(vrsSlug)
+            }.map {
+                val allocatedVerses = preallocateVerses(it, workbook.target, chapterNumber, draftNumber)
+                val versesFromText = getVersesFromText(projectSlug, chapterNumber, draftNumber)
+                val finalizedVerses = overlayVerses(allocatedVerses, versesFromText)
+                chunkCreator(finalizedVerses)
+            }
+            .subscribeOn(Schedulers.io())
+            .subscribe()
```

Comment:
sounds like this is more like an async method that will execute and complete at a later time when invoked. Should we documentation this so that the client code doesn't expect a "blocking" call?